docs/README.md

```diff
@@ -0,0 +1,114 @@
+* [`DelegationManager.deregisterOperatorFromAVS`][core-deregisterFromAVS]
+
+These methods ensure that the operator registering with the AVS is also registered as an operator in EigenLayer core. In this repo, these methods are called by the `RegistryCoordinator`.
+
+Eventually, operator slashing and payment for services will be part of the middleware/core relationship, but these features aren't implemented yet and their design is a work in progress.
+
+### System Components
+
+#### Registries
+
+| Code | Type | Proxy |
+| -------- | -------- | -------- |
+| [`RegistryCoordinator.sol`](../src/RegistryCoordinator.sol) | Singleton | Transparent proxy |
+| [`BLSApkRegistry.sol`](../src/BLSApkRegistry.sol) | Singleton | Transparent proxy |
+| [`StakeRegistry.sol`](../src/StakeRegistry.sol) | Singleton | Transparent proxy |
+| [`IndexRegistry.sol`](../src/IndexRegistry.sol) | Singleton | Transparent proxy |
+
+The `RegistryCoordinator` is the primary entry point for operators as they register for and deregister from an AVS's quorums. When operators register or deregister, the registry coordinator updates that operator's currently-registered quorums, and pushes the registration/deregistration to each of the three registries it controls:
+* `BLSApkRegistry`: tracks the aggregate BLS pubkey hash for the operators registered to each quorum. Also maintains a history of these aggregate pubkey hashes.
+* `StakeRegistry`: interfaces with the EigenLayer core contracts to track historical state of operators per quorum.
+* `IndexRegistry`: assigns indices to operators within each quorum, and tracks historical indices and operators per quorum. Used primarily by offchain infrastructure to fetch ordered lists of operators in quorums.
+
+Both the registry coordinator and each of the registries maintain historical state for the specific information they track. This historical state tracking can be used to query state at a particular block, which is primarily used in offchain infrastructure.
+
+See full documentation for the registry coordinator in [`RegistryCoordinator.md`](./RegistryCoordinator.md), and for each registry in [`registries/`](./registries/).
+
+#### BLSSignatureChecker
+
+| Code | Type | Proxy |
+| -------- | -------- | -------- |
+| [`BLSSignatureChecker.sol`](../src/BLSSignatureChecker.sol) | Singleton | Transparent proxy |
+| [`OperatorStateRetriever.sol`](../src/OperatorStateRetriever.sol) | Singleton | Transparent proxy |
+
+The BLSSignatureChecker verifies signatures made by the aggregate pubkeys ("Apk") of operators in one or more quorums. The primary function, `checkSignatures`, is called by an AVS when confirming that a given message hash is signed by operators belonging to one or more quorums.
+
+The `OperatorStateRetriever` is used by offchain code to query the `RegistryCoordinator` (and its registries) for information that will ultimately be passed into `BLSSignatureChecker.checkSignatures`.
```

Comment:
I believe this was designed to be re-useable across AVSs (unlike the other contracts)
I saw that the deployment info was deleted from the main README in this PR, which seems fine, but you might want to add that back here instead, so each AVS doesn't deploy a duplicate (or think they need to)

Reply:
Huh, I didn't see there was deployment info in the main README. I'll re-add that.

I also want the deployment info for the other contracts but left a TODO there since idk where that lives

Reply:
I'm fine with mentioning that the contract can be reused by other AVSs, but it's super super specific to our setup given all the specific registry functions it calls

And I'm not sure how much trouble that actually saves an AVS, given it's just a contract deployment and they could deploy a shallow clone instead (and not have to rely on us not rugging a multisig or something)

Reply:
[ac38452](https://github.com/Layr-Labs/eigenlayer-middleware/pull/96/commits/ac3845212bca0232f3db26882c12568b4a6e3660)

i kept the deploy info blank for the time being, so technically this is still TODO. i need more addresses - proxies and other.